engine/runtime/src/main/java/org/enso/interpreter/runtime/Module.java

```diff
@@ -589,19 +621,43 @@ private static Object evalExpression(
         ModuleScope scope, Object[] args, EnsoContext context, CallOptimiserNode callOptimiserNode)
         throws ArityException, UnsupportedTypeException {
       String expr = Types.extractArguments(args, String.class);
+
+      return doEval(context, callOptimiserNode, scope, LocalScope.root(), null, expr);
+    }
+
+    private static Object evalExpressionLocalScope(
+        ModuleScope scope, Object[] args, EnsoContext context, CallOptimiserNode callOptimiserNode)
+        throws ArityException, UnsupportedTypeException {
+      String expr = Types.extractArgument(args, String.class, 0);
+      LocalScope localScope = Types.extractArgument(args, LocalScope.class, 1);
+      VirtualFrame virtualFrame = Types.extractArgument(args, VirtualFrame.class, 2);
+
+      return doEval(context, callOptimiserNode, scope, localScope, virtualFrame, expr);
+    }
+
+    private static Object doEval(
```

Comment:
We shall not be modifying `Module` at all. The instrument shall use `DebugNode.doUncached` to evaluate the expression in the local scope itself.